matiasbenedetto/google-fonts-to-wordpress-collection
# Add 6.7 release

## What?
1. Add updated font collection for WordPress 6.7 release.
- Font collection with preview URLs.
- New SVG previews.
2. Improve the end stats of the preview creation script.

## Why?
Fixes: https://github.com/WordPress/google-fonts-to-wordpress-collection/issues/34

## How to test?
An easy way to test is to add this snippet to your theme `functions.php` file.
I think the most important thing to verify is:
- Where the [new fonts](https://github.com/WordPress/google-fonts-to-wordpress-collection/issues/34#issuecomment-2422720865) added? Search for 'Reddit' for example.
- Do the previews of those fonts look fine?
- Do the installation process continue working fine? Install a few fonts to test.

```php

function test_updated_font_collection() {
	// Unregister the default google fonts collection
	wp_unregister_font_collection( 'google-fonts' );

	// Register the Google fonts collection but with the new data for the 6.7 release
	wp_register_font_collection(
	  'google-fonts-67',
	  array(
	          // Changing the name of the collection just for testing purposes to see if the changes are working.
		  'name'          => _x( 'Google Fonts 6.7', 'font collection name' ),
		  'description'   => __( 'Install from Google Fonts. Fonts are copied to and served from your site.' ),
		  // Use the new collection data added in https://github.com/WordPress/google-fonts-to-wordpress-collection/pull/38
		  'font_families' => 'https://raw.githubusercontent.com/WordPress/google-fonts-to-wordpress-collection/6bfe35631278c414764b0dd50d4b1a8efe0e1e17/releases/wp-6.7/collections/google-fonts-with-preview.json',
		  'categories'    => array(
			  array(
				  'name' => _x( 'Sans Serif', 'font category' ),
				  'slug' => 'sans-serif',
			  ),
			  array(
				  'name' => _x( 'Display', 'font category' ),
				  'slug' => 'display',
			  ),
			  array(
				  'name' => _x( 'Serif', 'font category' ),
				  'slug' => 'serif',
			  ),
			  array(
				  'name' => _x( 'Handwriting', 'font category' ),
				  'slug' => 'handwriting',
			  ),
			  array(
				  'name' => _x( 'Monospace', 'font category' ),
				  'slug' => 'monospace',
			  ),
		  ),
	  )
	);
}

add_action( 'init', 'test_updated_font_collection' );

```

## Screencast
A screencast showing the collection with the updated data working:

https://github.com/user-attachments/assets/2038611f-60f7-4e3e-98d5-48137b97d1af



## To do after this PR is approved:
1. After this is approved and before merging we need to replace the provisory github base url (`https://raw.githubusercontent.com/WordPress/google-fonts-to-wordpress-collection/9724ef4a38da729f770027359b7decec2fcd29f8/releases/`) by the wporg CDN url (`https://s.w.org/images/fonts/`) in the `google-fonts-with-preview.json` file.
2. Ask meta team to host the JSON file and the SVG previews in the wporg CDN.
3. Change the Google fonts collection url in wordpress-develop repo.
